### ISSUE TITLE
Autosubmit should add Co-authored-by to commit message.

### ISSUE THREAD

**Issue by ditman** [flutter/flutter#107971]
When merging a PR that has had more than one contributor, the Github UI adds every other author as `Co-authored-by` of that PR.

(Note that this is only extracting information from who pushed directly to the PR branch, AFAIK it doesn't backtrack into merge commits and stuff like that)

Here's relevant documentation: [Creating a commit with multiple authors](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors).

And here's an example of a multiple-author commit (merged manually):

* https://github.com/flutter/plugins/commit/e8b38f6e03d1fea2746669c6e654714eca582b6e

The autosubmit bot should add Co-Author information. It gives recognition to all the people involved in the commit (and it looks cool from the "history" view too!)

**Comment by keyonghan** [comment]
Seems the autosubmit bot is behaving no difference from a manual merge.
See https://github.com/flutter/plugins/pull/7109 which was landed by the bot and the commit did show multiple authors: https://github.com/flutter/plugins/commit/9c312d4d2f5fe37608e11fa68295ecb59b2a200e.

Shall we close this bug?

**Comment by ditman** [comment]
@keyonghan yes, this can be closed. I suspect this is actually a problem with (each) repo configuration:

<img width="563" alt="Screenshot 2023-07-27 at 11 44 27 AM" src="https://github.com/flutter/flutter/assets/1255594/7c359860-ec4d-4f9a-a54d-6007e0cef07d">

But I don't have access to check or fix the configuration of any of the repos where this would be interesting (at least: `flutter/packages`, `flutter/flutter` and `flutter/engine`)

**Comment by ditman** [comment]
Here:

https://github.com/flutter/cocoon/blob/9dee587da5a1b1aa365e10815984b5812d22233e/auto_submit/lib/service/validation_service.dart#L82

I suspect that if `commitMessage` is `null`, GH will fallback to the default commit message/title. [Docs](https://docs.github.com/en/free-pro-team@latest/rest/pulls/pulls?apiVersion=2022-11-28#merge-a-pull-request).

(But I've never tested this, I suspect the old bot had this behavior on accident, maybe because we didn't try to tweak the title to fix the "Reland" pattern)

**Comment by keyonghan** [comment]
Thanks for the info. Then we need to make sure
1) Default message set correctly from repo side @Hixie could you help confirm it's default to `pull request title and commit details` for flutter/engine/packages repos?
2) `commitMessage` is not `null`, which seems enforced from the new bot.

**Comment by ditman** [comment]
@keyonghan the "Default" behavior is what we used to have before. It's a combo of the other options. If there's only one commit, it uses "pull request title and description". If there's multiple commits, it uses "pull request title and commit details". IMO, that's OK as well (because the co-authored-by is added to the commit details by github)

**Comment by keyonghan** [comment]
Related issue https://github.com/flutter/flutter/issues/49388

**Comment by ditman** [comment]
Good points there, I just think that preserving co-authorship is cool :P (but if it can't be done smoothly: ignore)
